src/js/actions/dialog.js

```diff
@@ -38,14 +53,28 @@ define(function (require, exports) {
      */
     var openDialog = function (id, dismissalPolicy) {
         var payload = {
-            id: id,
-            dismissalPolicy: dismissalPolicy
-        };
+                id: id,
+                dismissalPolicy: dismissalPolicy
+            },
+            neverPropagatePolicy = new PointerEventPolicy(
+                adapterUI.policyAction.NEVER_PROPAGATE,
+                adapterOS.eventKind.LEFT_MOUSE_DOWN
+            ),
+            isModal = this.flux.store("dialog").isModalDialog(id),
+            dispatchPromise = this.dispatchAsync(events.dialog.OPEN_DIALOG, payload),
+            policyPromise = isModal ?
+                this.transfer(policy.addPointerPolicies, [neverPropagatePolicy]) :
+                Promise.resolve();
 
-        return this.dispatchAsync(events.dialog.OPEN_DIALOG, payload);
+        return Promise.join(dispatchPromise, policyPromise, function (dispatchResult, policyResult) {
+            if (isModal) {
```

Comment:
nit/suggestion: this might look better if you just attach this as a `.then()` of the policyPromise.  Along the same lines, I think this might be more readable (above), and avoid a never-used instantiation of the neverPropagatePolicy when not modal,  if you separate out one single branch for the `isModal` code path that sets up the policyPromise accordingly, or `resolve` if not.